examples/static/samples/files/amp_url_converter.js

```diff
@@ -35,23 +35,19 @@ class AmpUrlConverter {
       this.showError('Empty input');
       return;
     }
-    try {
-      const ampUrl = this.ampUrlFactory.createAmpUrl(urlString);
-      const proxyUrl = ampUrl.getProxyUrl();
-      this.showResult(
-        '<a href="' + proxyUrl + '" target="_parent">' + proxyUrl + '</a>'
-      );
-    } catch (e) {
-      this.showError('Invalid URL');
-    }
+    this.createCacheUrl('cdn.ampproject.org', urlString)
+      .then(this.showResult.bind(this))
+      .catch(this.showError.bind(this));
   }
 
-  showError(message) {
+  showError() {
     this.resultView.className = 'error';
-    this.resultView.innerHTML = message;
+    this.resultView.innerHTML = 'invalid URL';
   }
 
-  showResult(result) {
+  showResult(cacheUrl) {
+    const result =
+      '<a href="' + cacheUrl + '" target="_blanb">' + cacheUrl + '</a>';
```

Comment:
Using a template string could also make this easier to read. [Browser support seems okay](https://caniuse.com/#feat=template-literals):

```suggestion
      `<a href="${cacheUrl}" target="_blank">${cacheUrl}</a>`;
```

Reply:
I'll leave it as is to ensure maximum backward compatibility.